### PR TITLE
Remove plugin version validation in canary and dev builds

### PR DESCRIPTION
**Pull request by mattcompiles**
# ↪️ Pull Request

Remove plugin version validation in dev/canary/nightly builds as they don't use semver and will always fail. 

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "test:integration-ci": "yarn workspace @parcel/integration-tests test-ci",
     "test": "yarn test:unit && yarn test:integration",
     "dev:release": "lerna publish -y --canary --preid dev --dist-tag=dev --exact --force-publish=* --no-git-tag-version --no-push",
-    "canary:release": "lerna publish -y --canary --preid canary --dist-tag=canary --exact --force-publish=* --no-git-tag-version --no-push",
+    "canary:release": "PARCEL_CANARY_BUILD=true lerna publish -y --canary --preid canary --dist-tag=canary --exact --force-publish=* --no-git-tag-version --no-push",
```
Comment (mattcompiles) [inline package.json]: I'm assuming this will be forwarded to the build step?

Comment (mattcompiles) [reply package.json]: I tested this with a dev-release and it looks good.

**Review by MonicaOlejniczak — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -30,8 +30,8 @@
     "test:integration": "yarn workspace @parcel/integration-tests test",
     "test:integration-ci": "yarn workspace @parcel/integration-tests test-ci",
     "test": "yarn test:unit && yarn test:integration",
-    "dev:release": "lerna publish -y --canary --preid dev --dist-tag=dev --exact --force-publish=* --no-git-tag-version --no-push",
-    "canary:release": "lerna publish -y --canary --preid canary --dist-tag=canary --exact --force-publish=* --no-git-tag-version --no-push",
+    "dev:release": "SKIP_PLUGIN_COMPATIBILITY_CHECK=true lerna publish -y --canary --preid dev --dist-tag=dev --exact --force-publish=* --no-git-tag-version --no-push",
+    "canary:release": "SKIP_PLUGIN_COMPATIBILITY_CHECK=true lerna publish -y --canary --preid canary --dist-tag=canary --exact --force-publish=* --no-git-tag-version --no-push",
```
Comment (devongovett) [inline package.json:34]: wouldn't these need to be set during the build, not during publishing?

Comment (mattcompiles) [reply package.json:34]: This works because the actual build is triggered by the `prepublishOnly` npm hook

**Pull request merged**